orc8r/cloud/go/obsidian/swagger/combine_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+ Copyright 2020 The Magma Authors.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree.
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package swagger_test
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"magma/orc8r/cloud/go/obsidian/swagger"
+	"magma/orc8r/cloud/go/obsidian/swagger/protos"
+	"magma/orc8r/cloud/go/orc8r"
+	swagger_lib "magma/orc8r/cloud/go/swagger"
+	"magma/orc8r/cloud/go/test_utils"
+
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v2"
+)
+
+func Test_GetCommonSpec(t *testing.T) {
+	specPath := "/etc/magma/configs/orc8r/swagger_specs"
+	commonSpecDir := "/etc/magma/configs/orc8r/swagger_specs/common"
+	commonSpecFilePath := filepath.Join(commonSpecDir, "swagger-common.yml")
+
+	os.RemoveAll(specPath)
+	defer os.RemoveAll(specPath)
+
+	err := os.MkdirAll(commonSpecDir, os.ModePerm)
+	assert.NoError(t, err)
+
+	commonTag := swagger_lib.TagDefinition{Name: "Tag Common"}
+	commonSpec := swagger_lib.Spec{Tags: []swagger_lib.TagDefinition{commonTag}}
+	yamlCommon := marshalToYAML(t, commonSpec)
+
+	err = ioutil.WriteFile(commonSpecFilePath, []byte(yamlCommon), 0644)
+	assert.NoError(t, err)
+
+	actual, err := swagger.GetCommonSpec()
+	assert.NoError(t, err)
+	assert.Equal(t, yamlCommon, actual)
+}
+
+func Test_GetCombinedSwaggerSpecs(t *testing.T) {
```

Comment:
P.P.S. add a test case to handle when there are warnings

Reply:
Oh! How come? If we were to do that, we would be kind of testing the "correctness" of the Combine() function and that's not what this test is about right?

Reply:
I understand the sentiment to add a test case where len(yamlSpecs) == 0 aka no registered Servicers exists but adding a test case to check for warnings seems to be edging towards the "correctness" of combine() rather than the polling aspect.